core/base/src/TSystem.cxx

```diff
@@ -2855,39 +2861,44 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
    Bool_t flatBuildDir = (fAclicProperties & kFlatBuildDir) || (opt && strchr(opt,'-')!=nullptr);
 
    // if non-zero, build_loc indicates where to build the shared library.
+   const char *cwd = WorkingDirectory();
    TString build_loc = ExpandFileName(GetBuildDir());
    if (build_dir && strlen(build_dir)) build_loc = build_dir;
    if (build_loc == ".") {
-      build_loc = WorkingDirectory();
+      build_loc = cwd;
    } else if (build_loc.Length() && (!IsAbsoluteFileName(build_loc)) ) {
-      AssignAndDelete( build_loc , ConcatFileName( WorkingDirectory(), build_loc ) );
+      AssignAndDelete( build_loc , ConcatFileName( cwd, build_loc ) );
    }
 
    // Get the include directory list in the dir1:dir2:dir3 format
    // [Used for generating the .d file and to look for header files for
    // the linkdef file]
-   TString incPath = GetIncludePath(); // of the form -Idir1  -Idir2 -Idir3
-   incPath.Append(":").Prepend(" ");
-   if (gEnv) {
-      TString fromConfig = gEnv->GetValue("ACLiC.IncludePaths","");
-      incPath.Append(fromConfig);
-   }
-   incPath.ReplaceAll(" -I",":");       // of form :dir1 :dir2:dir3
-   auto posISysRoot = incPath.Index(" -isysroot \"");
-   if (posISysRoot != kNPOS) {
-      auto posISysRootEnd = incPath.Index('"', posISysRoot + 12);
-      if (posISysRootEnd != kNPOS) {
-         // NOTE: should probably just skip isysroot for dependency analysis.
-         // (And will, in the future - once we rely on compiler-generated .d files.)
-         incPath.Insert(posISysRootEnd - 1, "/usr/include/");
-         incPath.Replace(posISysRoot, 12, ":\"");
+   TString incPath;
+   {
+      R__LOCKGUARD(gGlobalMutex); // NEWLY_ADDED
```

Comment:
I assume that this protects `TSystem::GetIncludePath` which should then be made thread safe on its own.